server/background_tasks.go

```diff
@@ -334,3 +339,124 @@ func (s *Server) StartBackgroundTasks() error {
+			streamInfo, err := serv.memphisStreamInfo(STORAGE_UPDATES_SUBJ)
+			if err != nil {
+				serv.Errorf("ConsumeStorageMsgs: " + err.Error())
+				errs <- err
+				return
+			}
+
+			if streamInfo.State.Msgs == 0 {
+				ping = ping + (20 * time.Second)
+				continue
+			}
+			responseChan := make(chan StoredMsg)
+			subject := fmt.Sprintf(JSApiRequestNextT, STORAGE_UPDATES_SUBJ, durableName)
+			reply := durableName + "_reply"
+			amount := 1000
+			req := []byte(strconv.FormatUint(uint64(amount), 10))
+			sub, err := serv.subscribeOnGlobalAcc(reply, reply+"_sid", func(_ *client, subject, reply string, msg []byte) {
+				go func(respCh chan StoredMsg, subject, reply string, msg []byte) {
+					replySubjs := reply
+					rawTs := tokenAt(reply, 8)
+					seq, _, _ := ackReplyInfo(reply)
+					intTs, err := strconv.Atoi(rawTs)
+					if err != nil {
+						serv.Errorf("ConsumeStorageMsgs: " + err.Error())
+						errs <- err
+						return
+					}
+
+					dataFirstIdx := 0
+					dataLen := len(msg)
+					dataFirstIdx = getHdrLastIdxFromRaw(msg) + 1
+					if dataFirstIdx > len(msg)-len(CR_LF) {
+						s.Errorf("ConsumeStorageMsgs: memphis error parsing in station get messages")
+					}
+
+					dataLen = len(msg) - dataFirstIdx
+					dataLen -= len(CR_LF)
+					header := msg[:dataFirstIdx]
+					data := msg[dataFirstIdx : dataFirstIdx+dataLen]
+					respCh <- StoredMsg{
+						Subject:       subject,
+						Sequence:      uint64(seq),
+						Data:          data,
+						Header:        header,
+						Time:          time.Unix(0, int64(intTs)),
+						ReplySubjects: replySubjs,
+					}
+				}(responseChan, subject, reply, copyBytes(msg))
+			})
+			if err != nil {
+				serv.Errorf("ConsumeStorageMsgs: " + err.Error())
+				errs <- err
+				return
+			}
+
+			serv.sendInternalAccountMsgWithReply(serv.GlobalAccount(), subject, reply, nil, req, true)
+			timer := time.NewTimer(time.Duration(timeout))
+
+			go func() {
+				for i := 0; i < amount; i++ {
+					select {
+					case <-timer.C:
+						timer.Stop()
+						serv.unsubscribeOnGlobalAcc(sub)
+						err := s.uploadToS3Storage(msgs)
+						if err != nil {
+							serv.Errorf("ConsumeStorageMsgs: " + err.Error())
+							errs <- err
+							break
+						}
+						// ack
+						for _, msg := range msgs {
+							reply := msg.ReplySubjects
+							serv.sendInternalAccountMsg(serv.GlobalAccount(), reply, []byte(_EMPTY_))
+						}
+					case msg := <-responseChan:
+						msgs = append(msgs, msg)
+					}
+				}
+			}()
+
+		}
+		time.Sleep(5 * time.Second)
+	}
+}
+
+func (s *Server) ListenForTierStorageMessages() error {
+
+	durableName := "storage_consumer"
+	cc := ConsumerConfig{
+		DeliverPolicy: DeliverAll,
+		AckPolicy:     AckExplicit,
+		Durable:       durableName,
+		FilterSubject: "$memphis_tiered_storage.>",
+		AckWait:       time.Duration(configuration.MAX_ACK_TIME_SECONDS_STORAGE) * time.Second,
+	}
+	err := serv.memphisAddConsumer(STORAGE_UPDATES_SUBJ, &cc)
+	if err != nil {
```

Comment:
think of it that 1 broker actually created the consumer the other 2 won't be able to create it since it already exists. or case in which you started the broker the consumer created and then you restart the broker. the creation will fail since it already exists - so you need to ignore an error the consumer already exists